### PR TITLE
[DOCS] Document static/dynamic watcher settings

### PR DESCRIPTION
**Pull request by jrodewig**
Indicates whether watcher settings are static or dynamic.

### Preview
https://elasticsearch_62218.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/notification-settings.html

### CODE REVIEW

File: docs/reference/settings/notification-settings.asciidoc [hunk 1]
```diff
@@ -103,72 +113,89 @@ can specify the following email account attributes:
 --
 [[email-account-attributes]]
```
[the remark this answers is not in the document]
Comment (jbaiera) [reply docs/reference/settings/notification-settings.asciidoc:115]: Or rather, this setting is defined in the code near these other email settings, and I do not see it in the immediate file, so if it's documented somewhere more sane, then perfect.

File: docs/reference/settings/notification-settings.asciidoc [hunk 2]
```diff
@@ -245,31 +281,37 @@ via Slack. You can specify the following Slack account attributes:
...
+The icon to display in the Slack messages.
+Overrides the incoming webhook's configured
+icon. Accepts a public URL to an image.
+
+`message_defaults.text`::
+(<<static-cluster-setting,Static>>)
+The default message content.
+
+`message_defaults.attachment`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:309]: Same question here re: dynamic message_defaults

File: docs/reference/settings/notification-settings.asciidoc [hunk 3]
```diff
@@ -245,31 +281,37 @@ via Slack. You can specify the following Slack account attributes:
...
+want to send messages to.
+
+`message_defaults.icon`::
+(<<static-cluster-setting,Static>>)
+The icon to display in the Slack messages.
+Overrides the incoming webhook's configured
+icon. Accepts a public URL to an image.
+
+`message_defaults.text`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:305]: Same question here re: dynamic message_defaults

File: docs/reference/settings/notification-settings.asciidoc [hunk 4]
```diff
@@ -245,31 +281,37 @@ via Slack. You can specify the following Slack account attributes:
...
+The sender name to display in the
+Slack message. Defaults to the watch ID.
+
+`message_defaults.to`::
+(<<static-cluster-setting,Static>>)
+The default Slack channels or groups you
+want to send messages to.
+
+`message_defaults.icon`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:299]: Same question here re: dynamic message_defaults

File: docs/reference/settings/notification-settings.asciidoc [hunk 5]
```diff
@@ -245,31 +281,37 @@ via Slack. You can specify the following Slack account attributes:
...
+(<<secure-settings,Secure>>)
+The Incoming Webhook URL to use to post messages to Slack. Required.
+
+`message_defaults.from`::
+(<<static-cluster-setting,Static>>)
+The sender name to display in the
+Slack message. Defaults to the watch ID.
+
+`message_defaults.to`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:294]: Same question here re: dynamic message_defaults

File: docs/reference/settings/notification-settings.asciidoc [hunk 6]
```diff
@@ -245,31 +281,37 @@ via Slack. You can specify the following Slack account attributes:
...
-                                    enable you to create more richly-formatted messages.
-                                    Specified as an array as defined in the
-                                    https://api.slack.com/docs/attachments[
-                                   Slack attachments documentation].
+`secure_url`::
+(<<secure-settings,Secure>>)
+The Incoming Webhook URL to use to post messages to Slack. Required.
+
+`message_defaults.from`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:289]: `xpack.notification.slack.account.***.message_defaults` is listed as dynamic in the code. Is this from setting overridden to be static only?

File: docs/reference/settings/notification-settings.asciidoc [hunk 7]
```diff
@@ -286,19 +328,23 @@ issues in Jira. You can specify the following Jira account attributes:
...
+`secure_password`::
+(<<secure-settings,Secure>>)
+The password of the user to connect to the Jira Software server. Required.
 
-  `issue_defaults`;;
-  Default fields values for the issue created in Jira. See
-  <<jira-action-attributes>> for more information.
-  Optional.
+`issue_defaults`::
+(<<static-cluster-setting,Static>>)
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:344]: Similar to the message_defaults setting, this affix setting seems to be configured as dynamic.

File: docs/reference/settings/notification-settings.asciidoc [hunk 8]
```diff
@@ -315,40 +361,48 @@ Specifies account information for sending notifications
...
 
-`secure_service_api_key` (<<secure-settings,Secure>>);;
+`secure_service_api_key`::
+(<<secure-settings,Secure>>)
 The https://developer.pagerduty.com/documentation/rest/authentication[
 PagerDuty API key] to use to access PagerDuty. Required.
 --
 +
-`event_defaults`;;
+`event_defaults`::
```
Comment (jbaiera) [inline docs/reference/settings/notification-settings.asciidoc:375]: Should this be marked as dynamic?

Comment (jbaiera) [reply docs/reference/settings/notification-settings.asciidoc:375]: Or rather, this is a dynamic setting in the code, but the below values are marked as static

**Review by jbaiera — COMMENTED** [review]
This got lost in my sea of browser tabs, sorry for the long wait. I left some questions about some properties that seemed to me that they would be dynamic based on the code, but I haven't verified if that's actually the case yet.

**Review by jrodewig — COMMENTED** [review]
Thanks for looking at this @jbaiera. I particularly appreciate you catching the mixup regarding several of the action settings. Not sure what happened originally, but those should be fixed now.

I also added a few missing `default_account` settings and the missing `allow_http` setting for Jira.

This is ready for another look at your convenience. No rush. Thanks!

**Review by jbaiera — APPROVED** [review]
Thanks for the updates LGTM!

**Comment by jrodewig** [comment]
Thanks @jbaiera!

**Pull request merged**